test/python/pulse/test_instruction_schedule_map.py

```diff
@@ -171,37 +175,62 @@ def test_sequenced_parameterized_schedule(self):
         with self.assertRaises(PulseError):
             inst_map.get('inst_seq', 0, 1, 2, 3, P1=1)
 
-        sched = inst_map.get('inst_seq', 0, 1, 2, 3)
+        p3_expr = Parameter('p3')
+        p3_expr = p3_expr.bind({p3_expr: 3})
+
+        sched = inst_map.get('inst_seq', 0, 1, 2, p3_expr)
         self.assertEqual(sched.instructions[0][-1].phase, 1)
         self.assertEqual(sched.instructions[1][-1].phase, 2)
         self.assertEqual(sched.instructions[2][-1].phase, 3)
 
-        sched = inst_map.get('inst_seq', 0, P1=1, P2=2, P3=3)
+        sched = inst_map.get('inst_seq', 0, P1=1, P2=2, P3=p3_expr)
         self.assertEqual(sched.instructions[0][-1].phase, 1)
         self.assertEqual(sched.instructions[1][-1].phase, 2)
         self.assertEqual(sched.instructions[2][-1].phase, 3)
 
-        sched = inst_map.get('inst_seq', 0, 1, 2, P3=3)
+        sched = inst_map.get('inst_seq', 0, 1, 2, P3=p3_expr)
         self.assertEqual(sched.instructions[0][-1].phase, 1)
         self.assertEqual(sched.instructions[1][-1].phase, 2)
         self.assertEqual(sched.instructions[2][-1].phase, 3)
 
     def test_schedule_generator(self):
         """Test schedule generator functionalty."""
 
-        x_test = 10
-        amp_test = 1.0
+        dur_val = 10
+        amp = 1.0
+
+        def test_func(dur: int):
+            sched = Schedule()
+            sched += Play(library.constant(int(dur), amp), DriveChannel(0))
+            return sched
+
+        expected_sched = Schedule()
+        expected_sched += Play(library.constant(dur_val, amp), DriveChannel(0))
+
+        inst_map = InstructionScheduleMap()
+        inst_map.add('f', (0,), test_func)
+        self.assertEqual(inst_map.get('f', (0,), dur_val), expected_sched)
+
+        self.assertEqual(inst_map.get_parameters('f', (0,)), ('dur',))
+
+    def test_schedule_generator_supports_parameter_expressions(self):
+        """Test expression-based schedule generator functionalty."""
+
+        dur_val = 10
+        dur_expr = Parameter('t')
+        amp = 1.0
 
-        def test_func(x):
+        def test_func(dur: ParameterExpression):
+            dur_bound = dur.bind({dur: dur_val})
             sched = Schedule()
-            sched += Play(library.constant(int(x), amp_test), DriveChannel(0))
+            sched += Play(library.constant(int(float(dur_bound)), amp), DriveChannel(0))
             return sched
 
-        ref_sched = Schedule()
-        ref_sched += Play(library.constant(x_test, amp_test), DriveChannel(0))
+        expected_sched = Schedule()
+        expected_sched += Play(library.constant(dur_val, amp), DriveChannel(0))
 
         inst_map = InstructionScheduleMap()
         inst_map.add('f', (0,), test_func)
-        self.assertEqual(inst_map.get('f', (0,), x_test), ref_sched)
+        self.assertEqual(inst_map.get('f', (0,), dur_expr), expected_sched)
```

Comment:
In this testcase `dur_expr` is fixed by `dur_val` defined in outer scope of generator (so parameter is kind of static). I understand that we have lack of framework that manages unbound parameters in the instruction schedule map, but here you intend in future this will be supported and `dur_expr` will be able to be updated dynamically? i.e.

```python
inst_map.get('f', (0,), t=10)
inst_map.get('f', (0,), t=20)
inst_map.get('f', (0,), t=30)
```

Reply:
@nkanazawa1989 Ah yes, that should be possible. In fact, I’ll update the test case to demonstrate this, when I get a chance. Great thought! 👍

Reply:
@nkanazawa1989 Updated. Let me know your thoughts.

Reply:
Perhaps I misunderstood the purpose of this PR (in my example above it doesn't need to use `ParameterExpression` and we can just feed the parameter `t` into generator). Now I understand the intention of your test case. Anyways the code looks good now.